src/main.cpp

```diff
@@ -714,92 +510,131 @@ static int parse_mode(const char *val, size_t val_len, void *storage, size_t sto
     };
 
-    struct option_udp {
-        char *addr;
-        bool server;
-        unsigned long port;
-        char *msgIdFilter;
-    };
     static const ConfFile::OptionsTable option_table_udp[] = {
-        {"address", true, ConfFile::parse_str_dup, OPTIONS_TABLE_STRUCT_FIELD(option_udp, addr)},
-        {"mode", true, parse_mode, OPTIONS_TABLE_STRUCT_FIELD(option_udp, server)},
-        {"port", false, ConfFile::parse_ul, OPTIONS_TABLE_STRUCT_FIELD(option_udp, port)},
-        {"filter", false, ConfFile::parse_str_dup,
-         OPTIONS_TABLE_STRUCT_FIELD(option_udp, msgIdFilter)}, // legacy AllowMsgIdOut
-        {"AllowMsgIdOut", false, ConfFile::parse_str_dup,
-         OPTIONS_TABLE_STRUCT_FIELD(option_udp, msgIdFilter)},
+        {"address", true, ConfFile::parse_stdstring,
+         OPTIONS_TABLE_STRUCT_FIELD(UdpEndpointConfig, address)},
+        {"mode", true, parse_udp_mode, OPTIONS_TABLE_STRUCT_FIELD(UdpEndpointConfig, mode)},
+        {"port", false, ConfFile::parse_ul, OPTIONS_TABLE_STRUCT_FIELD(UdpEndpointConfig, port)},
+        {"filter", false, parse_uint8_vector,
+         OPTIONS_TABLE_STRUCT_FIELD(UdpEndpointConfig, allow_msg_id_out)}, // legacy AllowMsgIdOut
+        {"AllowMsgIdOut", false, parse_uint8_vector,
+         OPTIONS_TABLE_STRUCT_FIELD(UdpEndpointConfig, allow_msg_id_out)},
     };
 
-    struct option_tcp {
-        char *addr;
-        unsigned long port;
-        int timeout;
-        char *msgIdFilter;
-    };
     static const ConfFile::OptionsTable option_table_tcp[] = {
-        {"address", true, ConfFile::parse_str_dup, OPTIONS_TABLE_STRUCT_FIELD(option_tcp, addr)},
-        {"port", true, ConfFile::parse_ul, OPTIONS_TABLE_STRUCT_FIELD(option_tcp, port)},
-        {"RetryTimeout", false, ConfFile::parse_i, OPTIONS_TABLE_STRUCT_FIELD(option_tcp, timeout)},
-        {"AllowMsgIdOut", false, ConfFile::parse_str_dup,
-         OPTIONS_TABLE_STRUCT_FIELD(option_tcp, msgIdFilter)},
+        {"address", true, ConfFile::parse_stdstring,
+         OPTIONS_TABLE_STRUCT_FIELD(TcpEndpointConfig, address)},
+        {"port", true, ConfFile::parse_ul, OPTIONS_TABLE_STRUCT_FIELD(TcpEndpointConfig, port)},
+        {"RetryTimeout", false, ConfFile::parse_i,
+         OPTIONS_TABLE_STRUCT_FIELD(TcpEndpointConfig, retry_timeout)},
+        {"AllowMsgIdOut", false, parse_uint8_vector,
+         OPTIONS_TABLE_STRUCT_FIELD(TcpEndpointConfig, allow_msg_id_out)},
     };
 
-    ret = conf.extract_options("General", option_table, ARRAY_SIZE(option_table), &opt);
+    ret = conffile.extract_options("General", option_table, ARRAY_SIZE(option_table), &config);
+    if (ret < 0) {
+        return ret;
+    }
+
+    ret = conffile.extract_options("General", option_table_log, ARRAY_SIZE(option_table_log),
+                                   &config.log_config);
     if (ret < 0) {
         return ret;
     }
 
     iter = {};
     pattern = "uartendpoint *";
```

Comment:
this should probably be a constant in UartEndpoint

Reply:
Makes sense, right 😉